cached_proc_macro/src/cached.rs

```diff
@@ -247,7 +284,16 @@ pub fn cached(args: TokenStream, input: TokenStream) -> TokenStream {
         #set_cache_and_return
     };
 
-    let do_set_return_block = if args.sync_writes {
+    let do_set_return_block = if args.sync_writes_by_key {
```

Comment:
Merge these two branches in the if condition.
If you do all above, the body will be the same.